src/main/common/form/models/statementOfMeans/expensesAndIncome/transactionSource.ts

```
@@ -47,12 +60,12 @@ export default class TransactionSource {
   @IsDefined({message: ValidationErrors.withMessage(ValidationErrors.SCHEDULE_SELECT_AN_OPTION)})
     schedule: TransactionSchedule;
 
-  constructor(name?: string, amount?: number, schedule?: TransactionSchedule, income?: boolean, nameRequired?: boolean) {
-    this.name = name;
-    this.amount = amount;
-    this.schedule = schedule;
-    this.income = income;
-    this.nameRequired = nameRequired;
+  constructor(params: TransactionSourceParams) {
+    this.name = params.name;
```

Null check on params? i.e. params?.name ?

---

bad idea in my opinion. params should never be null or undefined. and according to the constructor it does not allow undefined params. if it is a null then it should throw an error which it will in this casse

---

Maybe add a unit test for that then? You may already have done so!

---

it won't allow undefined so can't put a unit test for that. i would expect an error so not catching it. just like we would with other object parameters.